src/analyzers/practice/resistor-color-duo/ResistorColorDuoSolution.ts

```diff
@@ -1111,6 +1126,18 @@ class Entry {
     logger.log(`~> constant is not optimal`)
     return false
   }
+
+  private hasConstantDefinedInBody(): ShouldDefineTopLevelConstant | undefined {
+    const localConstants = extractVariables(this.body).filter(
+      (constant) =>
+        constant.init?.type === AST_NODE_TYPES.ArrayExpression ||
+        constant.init?.type === AST_NODE_TYPES.ObjectExpression
+    )
+    if (localConstants.length) {
+      const nameOfFirstConstant = localConstants[0].name || 'COLORS'
+      return new ShouldDefineTopLevelConstant(nameOfFirstConstant, '...')
+    }
+  }
 }
```

Comment:
You may want to change this to:

```suggestion
  private constantDefinedInBody(): string | null {
    const localConstants = extractVariables(this.body).filter(
      (constant) =>
        constant.init?.type === AST_NODE_TYPES.ArrayExpression ||
        constant.init?.type === AST_NODE_TYPES.ObjectExpression
    )
    
    if (localConstants.length) {
      return localConstants[0].name || 'COLORS'
    }
    
    return null;
  }
}
```

That would work better with everything else you're doing.